app/domain/authentication/authn_gcp/authenticator.rb

```diff
@@ -5,27 +5,37 @@ module AuthnGcp
 
     Authenticator = CommandClass.new(
       dependencies: {
-        validate_resource_restrictions: ValidateResourceRestrictions.new,
+        validate_resource_restrictions: Authentication::Common::ValidateResourceRestrictions.new,
+        authentication_request_class:   AuthenticationRequest,
         logger:                         Rails.logger
       },
       inputs:       [:authenticator_input]
     ) do
 
       extend Forwardable
-      def_delegators :@authenticator_input, :authenticator_name, :account, :username, :credentials
+      def_delegators :@authenticator_input, :authenticator_name, :service_id, :account, :username, :credentials
 
       def call
-        validate_resource_restrictions
+        authentication_request = create_authentication_request
+        validate_resource_restrictions(authentication_request)
       end
 
       private
 
-      def validate_resource_restrictions
+      def create_authentication_request
+        @authentication_request_class.new(
```

Comment:
do we want to use memoization? this way you can remove the line `authentication_request = create_authentication_request` from the `call` method.

Reply:
you could also remove `authentication_request` argument from the `validate_resource_restrictions` method this way.

Reply:
I know, but I prefer to be explicit about the arguments of the method.
When it depends on input from another method, I want it to be explicit.
Also, I think the dependency between the steps is more obvious that way.

Reply:
but you don't need to refer to it as an input from another method. i think that we can still have the steps. something like:
```
def call
  create_authentication_request
  validate_resource_restrictions
end

private 

def create_authentication_request
  authentication_request
end

def authentication_request
  @authentication_request ||= ...
end

def validate_resource_restrictions
  # do something with authentication_request
end
```

This style matches the rest of our code

Reply:
Although I prefer the explicit way, I understand this style, and adhere to it.
Done.